### PR TITLE
add rp2040 to host example build

### PR DESCRIPTION
**Pull request by hathach**
**Describe the PR**
Add RP2040 to host example build. Also change `ifdef RP2040_USB_HOST_MODE` to `if TUSB_OPT_HOST_ENABLED` to be consistent with tinyusb host detection convention.

### CODE REVIEW

File: hw/bsp/rp2040/family.cmake [hunk 1]
```diff
@@ -13,8 +13,15 @@ set(SRC_TINYUSB
...
+	${TOP}/src/host/hub.c
+	${TOP}/src/host/usbh.c
+	${TOP}/src/host/usbh_control.c
+	${TOP}/src/class/cdc/cdc_host.c
+	${TOP}/src/class/hid/hid_host.c
+	${TOP}/src/class/msc/msc_host.c
 	${TOP}/src/portable/raspberrypi/${FAMILY}/rp2040_usb.c
+	${TOP}/src/portable/raspberrypi/${FAMILY}/dcd_rp2040.c
+	${TOP}/src/portable/raspberrypi/${FAMILY}/hcd_rp2040.c
 )
```
Comment (liamfraser) [inline hw/bsp/rp2040/family.cmake:25]: We have split our cmake into a device and host mode so you don't unnecessarily compile both.

File: examples/host/cdc_msc_hid/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  target_include_directories(${PROJECT} PUBLIC
+    ${CMAKE_CURRENT_SOURCE_DIR}/src
+  )
+
+  # Example defines
+  target_compile_definitions(${PROJECT} PUBLIC
+    CFG_TUSB_OS=OPT_OS_PICO
+  )
+
+  target_link_libraries(${PROJECT} pico_stdlib pico_fix_rp2040_usb_device_enumeration)
```
Comment (liamfraser) [inline examples/host/cdc_msc_hid/CMakeLists.txt:41]: The USB device enumeration fix only applies to device mode. Perhaps this should be linked in by family.cmake?

File: hw/bsp/rp2040/family.cmake (hunk 1 above)
Comment (liamfraser) [reply hw/bsp/rp2040/family.cmake:25]: See https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/tinyusb/CMakeLists.txt

Comment (hathach) [reply hw/bsp/rp2040/family.cmake:25]: That is ok though, tinyusb is designed to be all included and controlled by config file. To be honest that make my  example easier 😄

File: examples/host/cdc_msc_hid/CMakeLists.txt (hunk 2 above)
Comment (hathach) [reply examples/host/cdc_msc_hid/CMakeLists.txt:41]: It is ok as well, since the code is not run anyway, as above. There may be otg example later on

**Pull request merged**